File: src/main/java/com/wesker926/leetcode/algorithms/p0475/s1/Solution.java
package com.wesker926.leetcode.algorithms.p0475.s1;

import java.util.Arrays;

/**
 * @author wesker.gh
 * @date 2021/12/20
 * @description 双数组遍历
 */
public class Solution {
    public int findRadius(int[] houses, int[] heaters) {
        Arrays.sort(houses);
        Arrays.sort(heaters);

        int max = 0;
        for (int i = 0, j = 0; i < houses.length; i++) {
            for (; j < heaters.length - 1 && Math.abs(houses[i] - heaters[j]) >= Math.abs(houses[i] - heaters[j + 1]); j++) {
            }
            max = Math.max(max, Math.abs(houses[i] - heaters[j]));
        }
        return max;
    }
}

File: src/main/java/com/wesker926/leetcode/algorithms/p0475/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/12/20
// @description: 双数组遍历
package s1

import (
	"math"
	"sort"
)

func findRadius(houses []int, heaters []int) int {
	sort.Ints(houses)
	sort.Ints(heaters)

	max := 0
	for i, j := 0, 0; i < len(houses); i++ {
		a := math.Abs(float64(houses[i] - heaters[j]))
		for ; j < len(heaters)-1 && a >= math.Abs(float64(houses[i]-heaters[j+1])); j++ {
		}

		if int(a) > max {
			max = int(a)
		}
	}
	return max
}
